pyccel/stdlib/STC_Extensions/Dict_extensions.h

```diff
@@ -0,0 +1,35 @@
+#ifndef DICT_EXTENSIONS_H
+#define DICT_EXTENSIONS_H
```

Comment:
This file shouldn't have a header guard. A different version of this file needs to be generated for each type of dictionary. The header guard prevents this. You should be able to see this with the following example:

```python
def f(k : int):
    a = {1:1.0, 2:2.0}
    b = {1:True, 2: False}
    if b.pop(k):
        return a.pop(k)
    else:
        return 3.0
```